CHANGELOG.md

```diff
@@ -4,7 +4,215 @@ All notable changes to this project will be documented in this file.
+    derive_for_type(path = "xcm::v2::multilocation::Junctions", derive = "Clone"),
+    derive_for_type(path = "xcm::v2::junction::Junction", derive = "Clone"),
+    derive_for_type(path = "xcm::v2::NetworkId", derive = "Clone"),
+    derive_for_type(path = "xcm::v2::BodyId", derive = "Clone"),
+    derive_for_type(path = "xcm::v2::BodyPart", derive = "Clone"),
+    derive_for_type(
+        path = "bounded_collections::weak_bounded_vec::WeakBoundedVec",
+        derive = "Clone"
+    )
+)]
+```
+
+This is something we plan to address in the next version of Subxt.
+
+### Change SignedExtension matching logic ([#1283](https://github.com/paritytech/subxt/pull/1283))
+
+Before this release, each signed extension had a unique name (`SignedExtension::NAME`). We'd use this name to figure out which signed extensions to apply for a given chain inside the `signed_extensions::AnyOf` type.
+
+However, we recently ran into a new signed extension in Substrate called `SkipCheckIfFeeless`. This extension would wrap another signed extension, but maintained its own name. It has since been "hidden" from the public Substrate interface again, but a result of encountering this is that we have generalised the way that we "match" on signed extensions, so that we can be smarter about it going forwards.
+
+So now, for a given signed extension, we go from:
+
+```rust
+impl<T: Config> SignedExtension<T> for ChargeAssetTxPayment<T> {
+    const NAME: &'static str = "ChargeAssetTxPayment";
+    type Decoded = Self;
+}
+```
+
+To:
+
+```rust
+impl<T: Config> SignedExtension<T> for ChargeAssetTxPayment<T> {
+    type Decoded = Self;
+    fn matches(identifier: &str, type_id: u32, types: &PortableRegistry) -> bool {
+        identifier == "ChargeAssetTxPayment"
+    }
+}
+```
+
+On the whole, we continue matching by name, as in the example above, but this allows an author to inspect the type of the signed extension (and subtypes of it) too if they want the signed extension to match (and thus be used) only in certain cases.
+
+### Remove `wait_for_in_block` helper method ([#1237](https://github.com/paritytech/subxt/pull/1237))
+
+One can no longer use `tx.wait_for_in_block` to wait for a transaction to enter a block. The reason for this removal is that, especially when we migrate to the new `chainHead` APIs, we will no longer be able to reliably obtain any details about the block that the transaction made it into.
+
+In other words, the following sort of thing would often fail:
+
+```rust
+tx.wait_for_in_block()
+  .await?
+  .wait_for_success()
+  .await?;
+```
+
+The reason for this is that the block announced in the transaction status may not have been "pinned" yet in the new APIs. In the old APIs, errors would occasionally be encountered because the block announced may have been pruned by the time we ask for details for it. Overall; having an "unrealiable" higher level API felt like a potential foot gun.
```

Comment:
Nit:
```suggestion
The reason for this is that the block announced in the transaction status may not have been "pinned" yet in the new APIs. In the old APIs, errors would occasionally be encountered because the block announced may have been pruned by the time we ask for details for it. Overall; having an "unreliable" higher level API felt like a potential foot gun.
```